Multiple Features added

dehlen opened the pull request:
Hello,

I added some features to this repository. 
- Added a UITextField subclass to listen for the "delete backward" button
- Delete Tags with the keyboard "delete backward" button
- Specify a maximum number of tags that can be added
- Allow more characters (like - _ + @ $) etc

Please evaluate and consider adding this to your repository I added them because I had some requirements in an app and I liked the simplicity of your repo.

AntonKuznetsov commented:
Hello @dehlen 
Thank you for your contribution
I left a couple of comments, could you please have a look?

Also It would be nice to have either one single commit or commits with different commit messages (now they have same exact messages)

dehlen commented:
Ok let me update the code regarding your comments and clean up the commits.

dehlen commented:
Ok I added a check for maxTags != nil. Therefore if you do not specify a number you can add infinite tags. Also the bug concerning the deleteBackward of my UITextField subclass was fixed with iOS 8.3. See here: http://stackoverflow.com/questions/25371254/how-to-detect-delete-key-on-an-uitextfield-in-ios-8. I also added the pragma mark to reflect the different delegates used. I will push changes immediately.

dehlen commented:
Ok see my updated repo.

AntonKuznetsov commented:
Hello @dehlen 
Sorry for late reply
I've tested your branch and it doesn't work for me

There you are using this code to initialise tags control:

```
    demoTagsControl = [[TLTagsControl alloc] initWithFrame:CGRectMake(8, 340, self.view.frame.size.width - 16, 36)
                                                  andTags:@[@"These", @"Tags", @"Are", @"Tapable"]
                                      withTagsControlMode:TLTagsControlModeList];
```

But this call is not defined in TLTagsControl.h

Also why do you write

```
@synthesize tapDelegate;
```

This is not necessary now.

Please test well
